src/interface/src/styleguide/expander-section/expander-section.component.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+import { NgClass, NgForOf, NgIf } from '@angular/common';
+
+export type OptionType = { value: string | number; text: string } | string;
+
+@Component({
+  selector: 'sg-expander-section',
+  standalone: true,
+  imports: [NgForOf, NgIf, NgClass],
+  templateUrl: './expander-section.component.html',
+  styleUrl: './expander-section.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ExpanderSectionComponent<T extends OptionType> {
+  @Input() isOpen?: boolean;
+  @Input() title: string = '';
+  @Input() groupName = '';
+  @Input() options: T[] = [];
+  @Input() defaultOption?: T;
+  @Output() optionSelected = new EventEmitter<T>();
```

Comment:
can we add comments to these? e.g., I'm not sure I fully understand the intent of groupName

Reply:
Ah sorry, I was reviewing this like it was out of draft -- no worries, add them when you're ready :D